### DISCUSSION TITLE
Fill the area with the nearest adjacent color

### DISCUSSION THREAD

**Discussion by thenamelessthing** [ImageMagick/ImageMagick#5704] [Help]
Hello everyone, I have an image that I want to print and so I add "bleed" to it with the following command:

`for file in proxies/*; do convert -background "#0c0f10" -transparent "#ffffff" "$file" -extent 1060x1458-36-36 "bleed_$file"; done`

However, once this command is launched I still have the 4 white corners between the bleed and the image that I would like to fill with the same color as the bleed. Surely there is a way to do this?

Here is the image before adding the bleed and the image with the bleed.

![Adunaphel](https://user-images.githubusercontent.com/10235448/198917559-80cc1d02-b378-45a7-9a50-fb10cf626586.jpg)

![Adunaphel](https://user-images.githubusercontent.com/10235448/198917573-e3fc5d2f-f553-447a-87c5-12e0dda208d7.jpg)

Thank you for your help!

**Comment by snibgo** [comment]
What version of IM, on what platform?

If your IM is v7, I suggest you use "magick" instead of "convert".

Your command is in a strange order. You should read the input, then do your processing, then write the output.

Your input is CMYK. Do you want that for the output, or do you want sRGB, or something else?

Your white corners are not exactly white, possibly because of JPEG compression. So you need a "-fuzz".

**Comment by thenamelessthing** [comment]
I'm using version 6.9 on ubuntu 20.04.

For the CMYK or SRGB, I don't know yet.

**Comment by thenamelessthing** [comment]
I want to keep CMYK because the result will be printed.

I have modified a little bit the command I use but I still have an issue.

`do convert -colorspace cmyk -crop 940x1338+24+24 -background "#0c0f10" -transparent "#ffffff" "$file" -extent 1110x1530-84-88 "bleed_$file"; done`

With the crop I can remove the white area on each corner of the card. After that, I increase the size of the area and fill the empty space with the black color of the border of the card. The problem is the result doesn'T give the correct black. A kind of dark grey. Any idea?


![image](https://user-images.githubusercontent.com/10235448/199382176-3f86947a-bfb0-47f6-9ec0-842f99678e6c.png)
